--- a/zlib.pc.in
+++ b/zlib.pc.in
@@ -1,7 +1,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
-sharedlibdir=@sharedlibdir@
+sharedlibdir=${libdir}
 includedir=@includedir@
 
 Name: zlib
--- a/zlib.pc.cmakein
+++ b/zlib.pc.cmakein
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@
 libdir=${exec_prefix}/lib
-sharedlibdir=${exec_prefix}/lib
+sharedlibdir=${libdir}
 includedir=${prefix}/include
 
 Name: zlib
