From c8ad6ac1d1ac7600138b9dbd0b92e88c36cc999c Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@altlinux.org>
Date: Tue, 6 Feb 2018 09:31:30 +0000
Subject: [PATCH 003/174] NEWS: add an entry for bug 22638

---
 NEWS | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/NEWS b/NEWS
index 0f7b0d5a9a..4bb08b1dbf 100644
--- a/NEWS
+++ b/NEWS
@@ -4,6 +4,14 @@ See the end for copying conditions.
 
 Please send GNU C library bug reports via <https://sourceware.org/bugzilla/>
 using `glibc' in the "product" field.
+
+Version 2.27.1
+
+The following bugs are resolved with this release:
+
+  [22638] sparc: static binaries are broken if glibc is built by gcc
+    configured with --enable-default-pie
+
 
 Version 2.27
 
-- 
2.30.2

