From 79543e547945085041e004b460c61135890a2607 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@altlinux.org>
Date: Tue, 6 Feb 2018 09:31:30 +0000
Subject: [PATCH 008/174] NEWS: add an entry for bug 22827

---
 NEWS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/NEWS b/NEWS
index 1e4cb10612..40c0fbc3ba 100644
--- a/NEWS
+++ b/NEWS
@@ -12,6 +12,7 @@ The following bugs are resolved with this release:
   [22638] sparc: static binaries are broken if glibc is built by gcc
     configured with --enable-default-pie
   [22797] Linux: use reserved name __key in pkey_get
+  [22827] RISC-V ELF64 parser mis-reads flag in ldconfig
 
 
 Version 2.27
-- 
2.30.2

