From 902af2f5eee71c3e48fe30d43fd7c61d563e975b Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Thu, 27 Jan 2022 12:20:21 -0800
Subject: [PATCH 096/247] NEWS: Add a bug fix entry for BZ #27457

---
 NEWS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/NEWS b/NEWS
index 09bb7dff66..bbec5e487a 100644
--- a/NEWS
+++ b/NEWS
@@ -44,6 +44,7 @@ The following bugs are resolved with this release:
   [23462] Static binary with dynamic string tokens ($LIB, $PLATFORM, $ORIGIN)
     crashes
   [27304] pthread_cond_destroy does not pass private flag to futex system calls
+  [27457] vzeroupper use in AVX2 multiarch string functions cause HTM aborts
   [27537] test-container: Always copy test-specific support files
   [27577] elf/ld.so --help doesn't work
   [27646] gethostbyname and NSS crashes after dlmopen
-- 
2.30.2

