From ceed89d0892a8940d9de7cff6654d5f2951714ea Mon Sep 17 00:00:00 2001
From: Carlos O'Donell <carlos@redhat.com>
Date: Tue, 12 Apr 2022 13:49:31 -0400
Subject: [PATCH 161/423] NEWS: Update fixed bug list for LD_AUDIT backports.

---
 NEWS | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/NEWS b/NEWS
index 187117f8a6..96e0589b1d 100644
--- a/NEWS
+++ b/NEWS
@@ -44,9 +44,18 @@ Security related changes:
 The following bugs are resolved with this release:
 
   [12889] nptl: Fix race between pthread_kill and thread exit
+  [15533] dynamic-link: LD_AUDIT introduces an avoidable performance
+    degradation
   [19193] nptl: pthread_kill, pthread_cancel should not fail after exit
   [22542] CVE-2022-23219: Buffer overflow in sunrpc clnt_create for "unix"
+  [23734] la_symbind*() doesn't get called when LD_BIND_NOW is set
+  [26643] register x8 and quad sized NEON registers are not properly
+    saved when using ld_audit on aarch64
   [28036] Incorrect types for pthread_mutexattr_set/getrobust_np
+  [28061] dynamic-link: A failing dlmopen called by an auditor crashed
+  [28062] dynamic-link: Suppress audit calls when a (new) namespace is
+    empty
+  [28096] elf: audit calls that uses static tls might fail
   [28182] _TIME_BITS=64 in C++ has issues with fcntl, ioctl, prctl
   [28223] mips: clone does not align stack
   [28310] Do not use affinity mask for sysconf (_SC_NPROCESSORS_CONF)
@@ -76,6 +85,7 @@ The following bugs are resolved with this release:
   [28770] CVE-2021-3998: Unexpected return value from realpath() for too long results
   [28784] x86: crash in 32bit memset-sse2.s when the cache size can not be determined
   [28850] linux: __get_nprocs_sched reads uninitialized memory from the stack
+  [28857] FAIL: elf/tst-audit24a
   [28860] build: --enable-kernel=5.1.0 build fails because of missing
     __convert_scm_timestamps
   [28865] linux: _SC_NPROCESSORS_CONF and _SC_NPROCESSORS_ONLN are inaccurate
-- 
2.30.2

