From cad3adf4ddeada37912c1c13b59a2ea5dd5d2832 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Tue, 23 May 2023 16:46:54 -0700
Subject: [PATCH 400/423] Document BZ #20975 fix

---
 NEWS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/NEWS b/NEWS
index 6fc6c27db5..d5bfe718e7 100644
--- a/NEWS
+++ b/NEWS
@@ -55,6 +55,7 @@ The following bugs are resolved with this release:
   [15533] dynamic-link: LD_AUDIT introduces an avoidable performance
     degradation
   [19193] nptl: pthread_kill, pthread_cancel should not fail after exit
+  [20975] Deferred cancellation triggers in __check_pf and looses lock leading to deadlock
   [22542] CVE-2022-23219: Buffer overflow in sunrpc clnt_create for "unix"
   [23734] la_symbind*() doesn't get called when LD_BIND_NOW is set
   [24816] Fix tst-nss-files-hosts-long on single-stack hosts
-- 
2.30.2

